Mathlib/CategoryTheory/Products/Associator.lean

```diff
@@ -53,6 +54,61 @@ instance inverseAssociatorIsEquivalence : (inverseAssociator C D E).IsEquivalenc
   (by infer_instance : (associativity C D E).inverse.IsEquivalence)
 #align category_theory.prod.inverse_associator_is_equivalence CategoryTheory.prod.inverseAssociatorIsEquivalence
 
--- TODO unitors?
+/-- The left unitor functor `1 × C ⥤ C`
+-/
+@[simps]
+def leftUnitor : Discrete PUnit × C ⥤ C where
+  obj X := X.2
+  map := @fun _ _ f => f.2
+#align category_theory.prod.left_unitor CategoryTheory.prod.leftUnitor
+
+/-- The right unitor functor `C × 1 ⥤ C`
+-/
+@[simps]
+def rightUnitor : C × Discrete PUnit ⥤ C where
+  obj X := X.1
+  map := @fun _ _ f => f.1
+#align category_theory.prod.right_unitor CategoryTheory.prod.rightUnitor
+
+/-- The left inverse unitor `C ⥤ 1 × C`
+-/
+@[simps]
+def leftInverseUnitor : C ⥤ Discrete PUnit × C where
+  obj X := ⟨⟨PUnit.unit⟩, X⟩
+  map := @fun _ _ f =>  ⟨𝟙 _, f⟩
+#align category_theory.prod.left_inverse_unitor CategoryTheory.prod.leftInverseUnitor
+
+/-- The right inverse unitor `C ⥤ C × 1`
+-/
+@[simps]
+def rightInverseUnitor : C ⥤ C × Discrete PUnit where
+  obj X := ⟨X, ⟨PUnit.unit⟩⟩
+  map := @fun _ _ f =>  ⟨f, 𝟙 _⟩
+#align category_theory.prod.right_inverse_unitor CategoryTheory.prod.rightInverseUnitor
+
+/-- The equivalence of categories expressing left unity of products of categories.
+-/
+def leftUnity : Discrete PUnit × C ≌ C :=
+  Equivalence.mk (leftUnitor C) (leftInverseUnitor C)
+    (NatIso.ofComponents fun X => eqToIso (by simp))
```

Comment:
Also, why not use the `where` syntax:
```lean
def leftUnitEquivalence : Discrete PUnit × C ≌ C where
  functor := ...
  inverse := ...
  unitIso := ...
  counitIso := ...
```
That would be 4 lines of code instead of 2, but this is much clearer IMHO.